Fix colorbar ticks

cvanelteren opened the pull request:
Addresses #105 

When `extend != 'neither'` the additional `update_ticks` is creating ghosting on the ticks. I am not sure why this happened but the `update_ticks` command is overridden by Ultraplot but deprecated by mpl. Removing the function outright breaks code elsewhere that I currently do not fully understand. I would be more in favor of removing this additional function but for now we can fix this by removing this additional call. Unittest is added to safe guard against future changes.

beckermr reviewed (changes requested):
Let's add `baseline/*` to the gitignore.

cvanelteren commented:
I thought I fixed it but now I am seeing locally weird stuff again. Back to the drawing board.

cvanelteren commented:
I think I had some test code messing with the formatter. I fixed the issue by forcing the tickrange to be set by default. The code creates an internal formatter that somehow does not have a default range for their ticks which makes it plot the entire range. Within the context of `extend != neither` it puts the values past the range on the location of the min and max of the original capped range (with `extend == nether`). I will check later if this can be traced by the way mpl handles these things but for now this would work --> more comfortable with what is causing the error.

cvanelteren commented:
I rebased and cleaned up the history here so that future people can track the changes. The fix should be ok and set default behavior will still do the things above but for now its beer time.

beckermr reviewed (approved):
Test failure is expected.
